File: Codility/Codility-js/4-counting-elements/MissingInteger.js
function solution(A) {
    var arr = [];
    for (var i = 0; i < A.length; i++){
        arr[A[i]] = true;
    }
    for (var i = 1; i <= arr.length; i++){
        if(!arr[i]) return i;
    }
    return 1;
}


console.log(solution([-1,-2,-4]));

File: Codility/Codility-py/04-counting-elements/MissingInteger.py
# MissingInteger


def solution(a):
    a.sort()
    smallest = 1

    for i in range(0, len(a)):
        if a[i] == smallest:
            smallest += 1

    return smallest


# (100%) https://codility.com/demo/results/trainingEB94WH-98C/

print(solution([1, 3, 6, 4, 1, 2]))  # 5
print(solution([1, 2, 3]))  # 4
print(solution([-1, -2]))  # 1
print(solution([2]))  # 1
print(solution([0]))  # 1
